### PR TITLE
Removed all references of $globalRole and $contextRole

### PR DESCRIPTION
**Pull request by thehenrytsai**
The distinction of "global role" and "context role" is still useful when discussing placement and usage, but changed the code to go through the same path (except edge case handling).

Added a validation (and test) to make sure protocol structure always references declared record types.

This "should" wrap up the generalization of roles.

### CODE REVIEW

**Review by csuwildcat — APPROVED** [review]

File: tests/interfaces/protocols-configure.spec.ts [hunk 1]
```diff
@@ -106,64 +106,52 @@ describe('ProtocolsConfigure', () => {
     });
 
     describe('protocol definition validations', () => {
-      it('allows `role` actions that have protocol path to valid $globalRole records', async () => {
+      it('should not allow a record in protocol structure to reference an on-existent record type', async () => {
```
Comment (LiranCohen) [inline tests/interfaces/protocols-configure.spec.ts:109]:
```suggestion
      it('should not allow a record in protocol structure to reference a non-existent record type', async () => {
```

File: Q_AND_A.md [hunk 2]
```diff
@@ -101,6 +101,13 @@
   - `write` - allows a DID to create and update the record they have created
   - `update` - allows a DID to update a record, regardless of the initial author
 
+- What is the difference between the terms "global role" and "context role"?
+
+  (Last update: 2024/02/16)
+
+  The structure and usage of "global roles" and "context roles" are identical. The distinction lies in their placement within the protocol hierarchy: a "global role" is defined as a root level record without a parent record, whereas a "context role" has a parent record thus making it "contextual". This means that a protocol rule set can have access to all global role records/assignments, hence the "global" designation; conversely, access to context role records/assignment is restricted to the specific context relevant to the message being authorized.
+
+
```
Comment (LiranCohen) [inline Q_AND_A.md:110]: What's the purpose of having this now that we don't have $globalRole and $contextRole anymore?

Comment (LiranCohen) [reply Q_AND_A.md:110]: Leaving this for now to merge in PR, but making a note to discuss.

**Review by LiranCohen — APPROVED** [review]
